ddl/index.go

```diff
@@ -1870,7 +1870,7 @@ func getNextPartitionInfo(reorg *reorgInfo, t table.PartitionedTable, currPhysic
 		// During index re-creation, process data from partitions to be added
 		nextPartitionDefs = pi.AddingDefinitions
 	}
-	if nextPartitionDefs == nil {
+	if len(nextPartitionDefs) == 0 {
```

Comment:
Do we have a test to test it？ Like check `txn-merge` or `lightning-merge`.

Reply:
I think it is related to the BR restore:

https://github.com/pingcap/tidb/blob/7d7350f2b2947606ac9cfef6beb17417570cfb49/executor/brie.go#L531

In the `Clone()` function of `PartitionInfo`, it initializes an empty slice:
https://github.com/pingcap/tidb/blob/7d7350f2b2947606ac9cfef6beb17417570cfb49/parser/model/model.go#L1179

It is not easy to write a unit test for this... Maybe we need an integration test or e2e test.

Reply:
Updated. PTAL